packages/server/src/server.ts

```diff
@@ -1,445 +1,72 @@
-                req,
-            );
-
-            // for response logging
-            res.locals.didRenderEpic = !!response.data;
-            res.locals.clientName = tracking.clientName;
-            res.locals.epicTargeting = {
-                weeklyArticleHistory: (targeting.weeklyArticleHistory ?? [])
-                    .slice(0, 3)
-                    .map((c: WeeklyArticleLog) => JSON.stringify(c)),
-            };
-
-            res.send(response);
-        } catch (error) {
-            next(error);
-        }
-    },
-);
-
-app.post(
-    '/liveblog-epic',
-    async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-        try {
-            const epicType: EpicType = 'LIVEBLOG';
-
-            const { tracking, targeting } = req.body;
-            const params = getQueryParams(req.query);
-            const response = await buildEpicData(
-                tracking,
-                targeting,
-                epicType,
-                params,
-                baseUrl(req),
-                req,
-            );
-
-            // for response logging
-            res.locals.didRenderEpic = !!response.data;
-            res.locals.clientName = tracking.clientName;
-
-            res.send(response);
-        } catch (error) {
-            next(error);
-        }
-    },
-);
-
-app.post(
-    '/banner',
-    async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-        try {
-            const { tracking, targeting } = req.body;
-            const params = getQueryParams(req.query);
-
-            const response = await buildBannerData(tracking, targeting, params, req);
-
-            // for response logging
-            res.locals.didRenderBanner = !!response.data;
-            res.locals.clientName = tracking.clientName;
-            // be specific about which fields to log, to avoid accidentally logging inappropriate things in future
-            res.locals.bannerTargeting = {
-                shouldHideReaderRevenue: targeting.shouldHideReaderRevenue,
-                showSupportMessaging: targeting.showSupportMessaging,
-                alreadyVisitedCount: targeting.alreadyVisitedCount,
-                countryCode: targeting.countryCode,
-                engagementBannerLastClosedAt: targeting.engagementBannerLastClosedAt,
-                subscriptionBannerLastClosedAt: targeting.subscriptionBannerLastClosedAt,
-                isPaidContent: targeting.isPaidContent,
-            };
-
-            res.send(response);
-        } catch (error) {
-            next(error);
+import { logError } from './utils/logging';
+import { buildEpicRouter } from './api/epicRouter';
+import { buildBannerRouter } from './api/bannerRouter';
+import { buildHeaderRouter } from './api/headerRouter';
+import { buildAmpEpicRouter } from './api/ampEpicRouter';
+import { buildModulesRouter } from './api/modulesRouter';
+
+const buildApp = (): Promise<Express> => {
```

Comment:
this function builds the express app. It returns a promise because in future it will initialise async dependencies before the server becomes available

Reply:
👍  For this comment. I was about to ask this exact question! 😄